### PR TITLE
Interpolated motion parser (ANB+MSET)

### PR DESCRIPTION
**Pull request by Xeeynamo**
Following the amazing discoveries and documentation from Kenjiuno, here there is the implementation of the interpolated type of motions, found inside ANB and MSET files. This is a continuation of #293 .

Note that is pull request is meant to fully parse (read+write) those files, not to implement the animation itself in-game as it will come in a later pull request.

This pull request is still a work in progress because it's only have been tested with `P_EX100.MSET\A000` and there is no documentation yet. The documentation I am going to write is complementary to [Kenjiuno's docs](https://gitlab.com/kenjiuno/msetDoc) and it is not meant to replace that.

### CODE REVIEW

File: OpenKh.Kh2/Motion.cs [hunk 1]
```diff
@@ -66,10 +118,186 @@ public class RawMotion
...
+            [Data] public float Value { get; set; }
+
+            public override string ToString() =>
+                $"{BoneIndex} {Transforms[Channel]} {Value}";
+        }
+
+        public class BoneAnimationTable
+        {
+            [Data] public short JointIndex { get; set; }
+            [Data] public byte Channel { get; set; }
```
Comment (kenjiuno) [inline OpenKh.Kh2/Motion.cs:214]: This `Channel` is bit fielded according to revel8n's feedback:

```
    uint8  keyType  : 4;
    uint8  preType  : 2;
    uint8  postType : 2;
```

image

Comment (Xeeynamo) [reply OpenKh.Kh2/Motion.cs:214]: Ok thanks, I will update code and docs following your feedback. What is the purpose of KeyCount?

**Comment by Xeeynamo** [comment]
Both parser and documentation are now ready to be reviewed

**Comment by kenjiuno** [comment]
Hi, I have checked Motion usage with #311. And also docs seems to be ok to be merged 👍

**Review by kenjiuno — APPROVED** [review]

**Pull request merged**